src/commands.c

```diff
@@ -1134,97 +976,20 @@ IMPLEMENT_COMMAND (cookie)
 
     const gchar *command = argv_idx (argv, 0);
 
-#ifdef USE_WEBKIT2
     WebKitWebContext *context = webkit_web_view_get_context (uzbl.gui.web_view);
     WebKitCookieManager *manager = webkit_web_context_get_cookie_manager (context);
-#endif
 
     if (!g_strcmp0 (command, "add")) {
-#ifdef USE_WEBKIT2
         uzbl_debug ("Manual cookie additions are unsupported in WebKit2.\n");
-#else
-        ARG_CHECK (argv, 7);
-
-        /* Parse with same syntax as ADD_COOKIE event. */
-        gchar *host = argv_idx (argv, 1);
-        gchar *path = argv_idx (argv, 2);
-        gchar *name = argv_idx (argv, 3);
-        gchar *value = argv_idx (argv, 4);
-        gchar *scheme = argv_idx (argv, 5);
-        gchar *expires_arg = argv_idx (argv, 6);
-
-        gboolean secure = FALSE;
-        gboolean httponly = FALSE;
-        SoupDate *expires = NULL;
-
-        if (g_str_has_prefix (scheme, "http")) {
-            secure = (scheme[4] == 's');
-            httponly = g_str_has_prefix (scheme + 4 + secure, "Only");
-        }
-        if (*expires_arg) {
-            expires = soup_date_new_from_time_t (strtoul (expires_arg, NULL, 10));
-        }
-
-        /* Create new cookie. */
-        /* TODO: Add support for adding non-session cookies. */
-        static const int session_cookie = -1;
-        SoupCookie *cookie = soup_cookie_new (name, value, host, path, session_cookie);
-        soup_cookie_set_secure (cookie, secure);
-        soup_cookie_set_http_only (cookie, httponly);
-        if (expires) {
-            soup_cookie_set_expires (cookie, expires);
-        }
-
-        /* Add cookie to jar. */
-        uzbl.net.soup_cookie_jar->in_manual_add = 1;
-        soup_cookie_jar_add_cookie (SOUP_COOKIE_JAR (uzbl.net.soup_cookie_jar), cookie);
-        uzbl.net.soup_cookie_jar->in_manual_add = 0;
-
-        if (expires) {
-            soup_date_free (expires);
-        }
-#endif
     } else if (!g_strcmp0 (command, "delete")) {
-#ifdef USE_WEBKIT2
         uzbl_debug ("Manual cookie deletions are unsupported in WebKit2.\n");
```

Comment:
Ugh. We'll have to look into how to do this; uzbl won't be nearly as safe with its cookies in WebKit2 as-is. Probably a new issue.